### ISSUE TITLE
DTO Crashes since integrating 4.9.0

### ISSUE THREAD

**Issue by j-j-m** [GetStream/stream-chat-swift#1767]
## What did you do?
installed 4.9.0

## What did you expect to happen?
My app would continue working as normal

## What happened instead?
it intermittently crashes with this stack

<img width="463" alt="Screen Shot 2022-01-24 at 2 13 47 PM" src="https://user-images.githubusercontent.com/5327798/151051455-ea1acc93-d502-4f92-959d-fdf731592a2f.png">

## GetStream Environment
**GetStream Chat version:** 4.9.0
**GetStream Chat frameworks:** StreamChat
**iOS version:** 13+
**Swift version:** 5.5
**Xcode version:** 13.2.1
**Device:** all

## Additional context

I am using both 
```swift
ChatClient.eventsController().allEventsPublisher 
```
to observe updates to channels in my custom list UI
and 
```swift
CurrentUserContoller.currentUserChangePublisher
```
to observe my users extra data.

any pointers would be appreciated

**Comment by j-j-m** [comment]
was able to track it down to this. trying to map the controllers channel to their view models... not sure if that brings any more insight

<img width="765" alt="image" src="https://user-images.githubusercontent.com/5327798/151073356-e4d17dcc-f767-408c-894f-55bcc961a184.png">

**Comment by adamrushy** [comment]
Thanks for the report @j-j-m I have created a ticket internally so we can investigate this further. Quick question, do you have any reproduction steps?

**Comment by b-onc** [comment]
Hello @j-j-m ,

Can you share the `cid`s you're fetching and your app id (not api key)?

**Comment by j-j-m** [comment]
sorry I don't remember the cid's, but I was able to fix this in the end. 

basically how my app works is I collect a list of ids and then create this channel controller and synchronize... 
I found that the code was creating this channel controller with a new query with each element added to the array. 
maybe it was a data race issue or something, but debouncing the creation of  the channel controller alleviated the issue.
